### PR TITLE
NTP module update

### PR DESCRIPTION
**Pull request by plorinquer**
This PR fixes a potential infinite loop in NTPExtPacketListField::getfield(). Also, references to the 'Raw' class were replaced by conf.raw_layer.

### CODE REVIEW

File: scapy/layers/ntp.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def _ntp_dispatcher(payload):
             mode_mask = 0x07
             mode = first_byte & mode_mask
 
-            cls = get_cls(_ntp_cls_by_mode.get(mode, "Raw"), conf.raw_layer)
+            cls = get_cls(_ntp_cls_by_mode.get(mode), conf.raw_layer)
```
Comment (p-l-) [inline scapy/layers/ntp.py]: To be consistent with your previous change, you might want to remove `conf.raw_layer`?

File: scapy/layers/ntp.py [hunk 2]
```diff
@@ -394,6 +394,9 @@ def getfield(self, pkt, s):
             extensions_len = len(extensions)
             while extensions_len >= 16:
                 ext_len = struct.unpack("!H", extensions[2:4])[0]
+                ext_len = min(ext_len, extensions_len)
+                if ext_len < 1:
+                    ext_len = extensions_len
```
Comment (p-l-) [inline scapy/layers/ntp.py]: What special cases are you fixing here? Maybe it could be good to add one in the regression tests?

Comment (plorinquer) [inline scapy/layers/ntp.py]: You are right. I have added one regression test that should trigger an infinite loop with the current version of the code.

File: scapy/layers/ntp.py (hunk 1 above)
Comment (plorinquer) [inline scapy/layers/ntp.py]: Yes, thank you. It has been done in the previous commit.

File: scapy/layers/ntp.py (hunk 2 above)
Comment (p-l-) [inline scapy/layers/ntp.py]: Perfect, but you've also add tests for VXLAN, which are out of the scope of this PR. Is that what you wanted or is that a mistake?

Comment (plorinquer) [inline scapy/layers/ntp.py]: Sorry, this is a mistake. These tests come from PR #247 . This has been fixed in the latest commit.

**Pull request merged**